learningequality/kolibri
# [WIP] Nanobox configuration for Kolibri

### Summary


This PR is to add nanobox's configuration for Kolibri.

Instructions:
1. To test the production before deploying the server:
    1. Run `nanobox evar add dry-run KOLIBRI_HOME=/app/kolibri_home` to set the KOLIBRI_HOME environment variable
    2. Create a pex file based on your code by doing `make dist pex` or get the pex file from a pull request. Move the pex file to the folder `deploy/` and rename it to `kolibri.pex`.
    3. Run `nanobox deploy dry-run` and then go to the ip address that nanobox gives. (note: the browser may show that the site is not secure, please ignore and go to the site anyway. reference: https://docs.nanobox.io/workflow/dry-run/#ssl-tls-in-dry-run)

2. To test the deployment of the server:
    1. Please create an account on Nanobox and ask Aron or me to add you to the team account on Nanobox. 
    2. The app `kolibri-demo-testing` has already been configured with the environment variable `KOLIBRI_HOME`, so there is no need to add it again on your side.
    3. Run `nanobox remote add kolibri-demo-testing`
    4. Run `nanobox deploy kolibri-demo-testing` to deploy the app
    5. After deploying, you can go to the dashboard on Nanobox to check the deployment process and go to the site by clicking the `LIVE APP` button on the right of the dashboard.

3. To test the integration with buildkite:
    1. Create a new pipeline on Buildkite with the target github repository
    2. Add environment variables `NANOBOX_APP_NAME=kolibri-demo-testing` and `BUILDKITE_CLEAN_CHECKOUT=true`. 
If you need to run `upload_artifacts` as well, please also include other environment variables from the current *kolibri* pipeline such as `GOOGLE_APPLICATION_CREDENTIALS` and `GITHUB_ACCESS_TOKEN` there.
    3. Push code to a branch, and the deployment on Nanobox will be triggered. 
    4. Create a pull request, and the build will be run, but the deployment will not be triggered.
    5. Merge the pull request, and the deployment on Nanobox will be triggered.

    I'm also using the following configurations for the pipeline
    <img width="419" alt="screen shot 2018-03-16 at 10 54 52 am" src="https://user-images.githubusercontent.com/19424916/37536679-8db6a014-2908-11e8-924e-0eb8ce1fa18b.png">


> Note: Currently, `nanobox deploy` only gets called when pushing commits into a branch but not creating a pull request. This is because to create a remote app on Nanobox, we have to manually add it on the dashboard instead of doing it through the command line. So, if you want to use nanobox to test Kolibri for your pull request, then you can test it using `dry-run` locally by either creating your pex with `make dist pex` or getting the pex file from the pull request and then moving the pex file to **deploy/kolibri.pex**. Sorry that there is currently not a good way to deploy the app for each pull request using Nanobox. Aron and I thought about using Kubernetes to do it but haven't started to implement yet.


### Reviewer guidance


Please follow the instructions in the summary to test.

### References


…

----

### Contributor Checklist


### Reviewer Checklist
